### DISCUSSION TITLE
Lorem Ipsum

### DISCUSSION THREAD

**Discussion by giscus** [DeclanChidlow/vale.rocks#23] [Comments Archive]
# Lorem Ipsum

A page for testing and trialing formatting, features, and typography by seeing how they interact in complex arrangements to catch edge cases at scale.

https://vale.rocks/posts/lorem-ipsum

**Comment by DeclanChidlow** [comment]
# GitHub Flavored Markdown (GFM) Test Page

This page demonstrates all major features of GitHub Flavored Markdown provided via Giscus in the comments section of [Vale.Rocks](https://vale.rocks).

## Basic Formatting

Normal text. _Italic text._ **Bold text.** **_Bold italic text._** ~~Strikethrough text.~~ ~~_Italic strikethrough text._~~ ~~**Bold strikethrough text.**~~ ~~**_Bold italic strikethrough text._**~~ `Code text.` _`Code italic text.`_ **`Code bold text.`** **_`Code bold italic text.`_** ~~`Code strikethrough text.`~~ ~~**_`Code bold italic strikethrough text.`_**~~

Normal superscript<sup>sup</sup>. Normal subscript<sub>sub</sub>. _Italic superscript<sup>sup</sup>_. _Italic subscript<sub>sub</sub>._ **Bold superscript<sup>sup</sup>**. **Bold subscript<sub>sub</sub>.** **_Bold italic superscript<sup>sup</sup>_**. **_Bold italic subscript<sub>sub</sub>._** ~~Strikethrough superscript<sup>sub</sup>.~~ ~~Strikethrough subscript<sub>sub</sub>.~~ ~~**_Bold italic strikethrough superscript<sup>sup</sup>_**~~. ~~**_Bold italic strikethrough subscript<sub>sub</sub>._**~~

Superscript and subscript together<sup>sup</sup><sub>sub</sub>.

Here's a horizontal rule:

---

## Lists

### Unordered Lists

* Item 1
* Item 2
  * Nested item 2.1
  * Nested item 2.2
* Item 3

### Ordered Lists

1. First item
2. Second item
   1. Nested item 2.1
   2. Nested item 2.2
3. Third item

## Links and Images

### Links

- [Basic link to Vale's website](https://vale.rocks)
- [Link with title](https://vale.rocks "Vale's Website")
- <https://vale.rocks> (Automatic linking)
- Reference-style [link][1]

[1]: https://vale.rocks

### Images

![Flipper Zero on Table](https://vale.rocks/assets/posts/i-got-a-flipper-zero/flipper-zero-on-table.avif "Flipper Zero Device")

### Image with Link

[![Flipper Zero](https://vale.rocks/assets/posts/i-got-a-flipper-zero/flipper-zero-on-table.avif "Click to visit Vale's website")](https://vale.rocks)

## Code

### Inline Code

Use `console.log()` to output to the console.

### Code Blocks

```javascript
function greet(name) {
    console.log(`Hello, ${name}!`);
    return true;
}

// Call the function
greet("World");
```

```python
def factorial(n):
    if n == 0:
        return 1
    return n * factorial(n - 1)

# Calculate factorial of 5
print(factorial(5))
```

### Syntax Highlighting with Diff

```diff
- const oldFunction = (x) => x * 2;
+ const newFunction = (x) => x * 3;

- console.log(oldFunction(5));  // Output: 10
+ console.log(newFunction(5));  // Output: 15
```

## Tables

### Simple Table

| Header 1 | Header 2 | Header 3 |
|----------|----------|----------|
| Cell 1   | Cell 2   | Cell 3   |
| Cell 4   | Cell 5   | Cell 6   |

### Aligned Table

| Left-aligned | Center-aligned | Right-aligned |
|:-------------|:-------------:|-------------:|
| Left         | Center        | Right        |
| Text         | Text          | Text         |

### Complex Table

| Feature | Description | Support |
|---------|-------------|---------|
| Tables | GFM tables support | ✅ |
| Lists | Ordered and unordered | ✅ |
| Code blocks | With syntax highlighting | ✅ |
| Math | LaTeX-style equations | ✅ |

## Task Lists

- [x] Write the press release
- [ ] Update the website
- [x] Contact the media
- [ ] Prepare for launch
  - [x] Finish documentation
  - [ ] Test all features
  - [ ] Update changelog

## Mathematics

### Inline Math

The quadratic formula is $$x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}$$.

### Block Math

The Gaussian integral:

$$
\int_{-\infty}^{\infty} e^{-x^2} dx = \sqrt{\pi}
$$

The Euler identity:

$$
e^{i\pi} + 1 = 0
$$

## Extended Features

### Footnotes

Here's a sentence with a footnote[^1].

[^1]: This is the footnote content.

### Callouts/Alerts

> [!NOTE]
> Important info users should be aware of, even when skimming.

> [!TIP]
> Practical advice for improving or simplifying tasks.

> [!IMPORTANT]
> Critical info for achieving the desired outcome.

> [!WARNING]
> Urgent info requiring immediate attention to prevent issues.

> [!CAUTION]
> Alerts about potential risks or negative consequences.

### Keyboard Keys

Use <kbd>Ctrl</kbd> + <kbd>C</kbd> to copy text.

### Details/Summary (Collapsible Content)

<details>
<summary>Click to expand</summary>

This is hidden content that can be expanded by clicking.

1. Hidden item 1
2. Hidden item 2
3. Hidden item 3

</details>

### Emoji Support

:smile: :heart: :thumbsup: :rocket:

### Colours

`#0969DA` `rgb(9, 105, 218)` `hsl(212, 92%, 45%)`

---

## Conclusion

This document demonstrates the major features of GitHub Flavored Markdown. For more information, visit the [GitHub Markdown Guide](https://docs.github.com/en/github/writing-on-github).
